src/component/Tooltip.tsx

```diff
@@ -202,101 +196,120 @@ export class Tooltip<TValue extends ValueType, TName extends NameType> extends P
+    ...translateStyle({
+      transform: useTranslate3d
+        ? `translate3d(${translateX}px, ${translateY}px, 0)`
+        : `translate(${translateX}px, ${translateY}px)`,
+    }),
+    ...outerStyle,
+  };
 
+  if (isAnimationActive && active) {
     outerStyle = {
       ...translateStyle({
-        transform: this.props.useTranslate3d
-          ? `translate3d(${translateX}px, ${translateY}px, 0)`
-          : `translate(${translateX}px, ${translateY}px)`,
+        transition: `transform ${animationDuration}ms ${animationEasing}`,
       }),
       ...outerStyle,
     };
+  }
 
-    if (isAnimationActive && active) {
-      outerStyle = {
-        ...translateStyle({
-          transition: `transform ${animationDuration}ms ${animationEasing}`,
-        }),
-        ...outerStyle,
-      };
-    }
+  const cls = classNames(CLS_PREFIX, {
+    [`${CLS_PREFIX}-right`]: isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX >= coordinate.x,
+    [`${CLS_PREFIX}-left`]: isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX < coordinate.x,
+    [`${CLS_PREFIX}-bottom`]:
+      isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY >= coordinate.y,
+    [`${CLS_PREFIX}-top`]: isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY < coordinate.y,
+  });
 
-    const cls = classNames(CLS_PREFIX, {
-      [`${CLS_PREFIX}-right`]:
-        isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX >= coordinate.x,
-      [`${CLS_PREFIX}-left`]: isNumber(translateX) && coordinate && isNumber(coordinate.x) && translateX < coordinate.x,
-      [`${CLS_PREFIX}-bottom`]:
-        isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY >= coordinate.y,
-      [`${CLS_PREFIX}-top`]: isNumber(translateY) && coordinate && isNumber(coordinate.y) && translateY < coordinate.y,
-    });
+  return (
+    // ESLint is disabled to allow listening to the `Escape` key. Refer to
+    // https://github.com/recharts/recharts/pull/2925
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+    <div
+      role="dialog"
+      onKeyDown={event => {
+        if (event.key === 'Escape') {
+          setState(prev => ({
+            ...prev,
+            dismissed: true,
+            dismissedAtCoordinate: {
+              ...prev.dismissedAtCoordinate,
+              x: coordinate.x,
+              y: coordinate.y,
+            },
+          }));
+        }
+      }}
+      className={cls}
+      style={outerStyle}
+      ref={wrapperNode}
+    >
+      {renderContent(content, {
+        ...props,
+        payload: finalPayload,
+      })}
+    </div>
+  );
+};
 
-    return (
-      // ESLint is disabled to allow listening to the `Escape` key. Refer to
-      // https://github.com/recharts/recharts/pull/2925
-      // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
-      <div
-        tabIndex={-1}
```

Comment:
Why remove the tabIndex?

Reply:
This should be in a separate PR sorry - but imho it shouldn't be there. It was added with accessibility in mind and its only made accessibility worse by blurring focused elements (say in a form) on chart hover.

Will add it back for this PRs sake